.travis.yml

```diff
@@ -24,7 +24,7 @@ env:
     - MONGODB=3.2.20
     - MONGODB=3.4.16
     - MONGODB=3.6.6
-    - MONGODB=4.0.0
+    - MONGODB=4.0.12
```

Comment:
Unfortunately, due to the [license changes](https://www.mongodb.com/community/licensing) in MongoDB, I don't think we can do this at the moment. So I suggest that we revert this.

Reply:
@venilnoronha with some time past now and also seeing that spring-data-mongo uses 4.2 in their [Dockerfile](https://github.com/spring-projects/spring-data-mongodb/blob/master/ci/openjdk11-mongodb-4.2/Dockerfile) is this something you would reconsider integrating if I updated this pull request with the latest mongo 4.0 and 4.2 versions? 

Additionally I can update the other spring cloud versions to latest were applicable and address the 0.0.4 issue